docs/application/native/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,634 @@
+
+1. Declare a service-component in the manifest file:
+    ```
+    <component-based-application appid="org.tizen.base-component" exec="@BINDIR@/base-component" nodisplay="false" multiple="false" type="capp">
+        <label>Base-component application</label>
+        <icon>@DESKTOP_ICON@</icon>
+        <service-component id="base-service" launch_mode="caller" main="false" icon-display="false" taskmanage="true">
+            <icon>org.tizen.sample.png</icon>
+            <label>base-service</label>
+            <label xml:lang="en-us">base-service</label>
+            <label xml:lang="ko-kr">base-service[KOR]</label>
+        </service-component>
+    </component-based-application>
+    ```
+
+2. Add a service-component to the component-based application:
+    ```
+    component_class_h service_component_add(component_class_h comp_class,
+            const char *component_id, void *user_data)
+    {
+        service_component_lifecycle_callback_s callback = {
+            .create = __service_component_create_cb,
+            .start_command = __service_component_start_command_cb,
+            .destroy = __service_component_destroy_cb,
+            .restore_content = __service_component_restore_content_cb,
+            .save_content = __service_component_save_content_cb,
+            .action = __service_component_action_cb,
+            .device_orientation_changed = __service_component_device_orientation_changed_cb,
+            .language_changed = __service_component_language_changed_cb,
+            .region_format_changed = __service_component_region_format_changed_cb,
+            .low_battery = __service_component_low_battery_cb,
+            .low_memory = __service_component_low_memory_cb,
+            .suspended_state_changed = __service_component_suspended_state_changed_cb,
+        };
+
+        return component_based_app_add_service_component(comp_class,
+                component_id, &callback, user_data);
+    }
+
+    component_class_h __app_create_cb(void *user_data)
+    {
+         component_class_h comp_class = NULL;
+         comp_class = frame_component_add(comp_class, "base-frame", NULL);
+         comp_class = service_component_add(comp_class, "base-service", NULL);
+        return comp_class;
+    }
+
+    void __app_terminate_cb(void *user_data)
+    {
+    }
+
+    int main(int argc, char** argv)
+    {
+        int ret;
+        component_based_app_lifecycle_callback_s callback = {
+            .create = __app_create_cb,
+            .terminate = __app_terminate_cb
+        };
+
+        ret = component_based_app_main(argc, argv, &callback, NULL);
+        if (ret != APP_ERROR_NONE)
+            return ret;
+
+        return 0;
+    }
+    ```
+
+<a name="managing_action"></a>
+## Managing Actions
+
+The component-based application can register actions and receive action events. You can send an action event to another application using app-control.
```

Comment:
Global comment:
Please use the code tag for `app-control`.
Please fix this throughout the document.